### PR TITLE
refactor: Port LoreUtils to StyledText

### PR DESCRIPTION
**Pull request by Mcrtin**

### CODE REVIEW

File: common/src/main/java/com/wynntils/utils/mc/LoreUtils.java [hunk 1]
```diff
@@ -116,18 +76,11 @@ public static Matcher matchLoreLine(ItemStack itemStack, int startLineNum, Patte
...
+     * @return a {@link StyledText} representation of the lore
      */
-    public static CodedString getStringLore(ItemStack itemStack) {
-        StringBuilder toReturn = new StringBuilder();
-        for (CodedString x : getLore(itemStack)) {
-            toReturn.append(x);
-        }
-        return CodedString.fromString(toReturn.toString());
+    public static StyledText getStringLore(ItemStack itemStack) {
+        return StyledText.join("\n", getLore(itemStack));
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/utils/mc/LoreUtils.java]: This method doesn't do the same thing it did?  I also saw you changed how this method is used. Can you explain? (If there is not a good reason I might want this reverted, any unnecessary change in these kinds of PRs can lead to a big mess since this scale of ports is already bug-prone enough and hard to follow)

Comment (Mcrtin) [reply common/src/main/java/com/wynntils/utils/mc/LoreUtils.java]: you are right. It made more sense to me but could lead to many unforeseen interactions because it is also used in LoreUtils#getJsonFromIngameLore. I will drop this change.

File: common/src/main/java/com/wynntils/features/debug/LogItemInfoFeature.java [hunk 2]
```diff
@@ -70,7 +69,7 @@ private static String getDescription(ItemStack itemStack) {
 
         return "[Logging Item]\nName: "
                 + itemStack.getHoverName().getString() + "\nLore:\n"
-                + CodedString.join("\n", LoreUtils.getLore(itemStack)) + "\nItem Type: "
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/debug/LogItemInfoFeature.java:73]: Revert this as well?

File: common/src/main/java/com/wynntils/utils/mc/LoreUtils.java [hunk 3]
```diff
@@ -116,18 +76,11 @@ public static Matcher matchLoreLine(ItemStack itemStack, int startLineNum, Patte
     }
 
     /**
-     * Get the lore from an item, combined into one string.
-     * Relies on {@link #getLore(ItemStack)}. See the aforementioned
-     * for a list format
-     *
-     * @return a {@link String} containing all item lore
+     * Joins the lines of the lore of the given itemStack into a single one with new line characters.
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/utils/mc/LoreUtils.java]: Is this comment accurate?

**Review by kristofbolyai — COMMENTED** [review]
Mostly looks good!

**Review by kristofbolyai — APPROVED** [review]
Looks good! Thanks.
